File: PST/FUEL/src/FUEL/clsCreateKey.vb
Imports Microsoft.VisualBasic.CompilerServices
Imports System

Namespace FUEL
    Friend Class clsCreateKey
        ' Methods
        Public Shared Function GeneratePropertyKey(ByVal DateToExpire As DateTime) As String
            Return New Simple3Des("Overlanding123").EncryptData(Conversions.ToString(CDbl((Conversions.ToDouble(Conversions.ToString(DateToExpire.Ticks)) * 3.1415926535897931))))
        End Function

        Public Shared Function GenerateScriptKey(ByVal DateToExpire As DateTime) As String
            Return clsCreateKey.GeneratePropertyKey(DateToExpire)
        End Function

    End Class
End Namespace

File: PST/FUEL Source Code/FUEL/clsCreateKey.cs
// Decompiled with JetBrains decompiler
// Type: FUEL.clsCreateKey
// Assembly: FUEL, Version=2.4.8.0, Culture=neutral, PublicKeyToken=null
// MVID: A3D630D0-0620-425D-8CFB-B67B2AC75507
// Assembly location: C:\hp\FUEL\FUEL.dll

using Microsoft.VisualBasic.CompilerServices;
using System;
using System.Diagnostics;

namespace FUEL
{
  internal class clsCreateKey
  {
    [DebuggerNonUserCode]
    public clsCreateKey()
    {
    }

    public static string GeneratePropertyKey(DateTime DateToExpire) => new Simple3Des("Overlanding123").EncryptData(Conversions.ToString(Conversions.ToDouble(Conversions.ToString(DateToExpire.Ticks)) * Math.PI));

    public static string GenerateScriptKey(DateTime DateToExpire) => clsCreateKey.GeneratePropertyKey(DateToExpire);
  }
}
